test/class_test.js

```diff
@@ -93,6 +93,67 @@ describe('classes', () => {
     `);
   });
 
+  describe('assign properties from method parameters', () => {
+    it('constructor without function body', () => {
+      check(`
+        class A
+          constructor: ([@a = 1], {test: @b = 2}, @c) ->
+      `, `
+        class A {
+          constructor([a = 1], {test: b = 2}, c) {
+            this.a = a;
+            this.b = b;
+            this.c = c;
+          }
+        }
+      `);
+    });
+    it('constructor with function body', () => {
+      check(`
+        class A
+          constructor: (@a) ->
+            return
+      `, `
+        class A {
+          constructor(a) {
+            this.a = a;
+            return;
+          }
+        }
+      `);
+    });
+    it('method', () => {
+      check(`
+        class A
+          method: (@a) ->
+      `, `
+        class A {
+          method(a) {
+            this.a = a;
+            return;
+          }
+        }
+      `);
+    });
+    it('bound method', () => {
+      check(`
+        class A
+          method: (@a) =>
+      `, `
+        class A {
+          constructor() {
+            this.method = this.method.bind(this);
+          }
+        
+          method(a) {
+            this.a = a;
+            return;
```

Comment:
Another possibility is to use an identifier that we're pretty sure won't be actually returned. So use `return DECAFFEINATE_24820832` where the number is calculated once using `Date.now()` or something, then the main stage could look for such return statements and remove them.